src/main/java/org/jabref/logic/importer/fileformat/CffImporter.java

```diff
@@ -88,43 +110,99 @@ public CffIdentifier() {
         }
     }
 
+    private static class CffReference {
+        private final HashMap<String, String> values = new HashMap<>();
+
+        @JsonProperty("authors")
+        private List<CffEntity> authors;
+
+        @JsonProperty("conference")
+        private CffConference conference;
+
+        @JsonProperty("contact")
+        private CffEntity contact;
+
+        @JsonProperty("editors")
+        private List<CffEntity> editors;
+
+        @JsonProperty("editors-series")
+        private List<CffEntity> editorsSeries;
+
+        @JsonProperty("database-provider")
+        private CffEntity databaseProvider;
+
+        @JsonProperty("institution")
+        private CffEntity institution;
+
+        @JsonProperty("keywords")
+        private List<String> keywords;
+
+        @JsonProperty("languages")
+        private List<String> languages;
+
+        @JsonProperty("location")
+        private CffEntity location;
+
+        @JsonProperty("publisher")
+        private CffEntity publisher;
+
+        @JsonProperty("recipients")
+        private List<CffEntity> recipients;
+
+        @JsonProperty("senders")
+        private List<CffEntity> senders;
+
+        @JsonProperty("translators")
+        private List<CffEntity> translators;
+
+        @JsonProperty("type")
+        private String type;
+
+        public CffReference() {
+        }
+
+        @JsonAnySetter
+        private void setValues(String key, String value) {
+            values.put(key, value);
+        }
+    }
+
     @Override
     public ParserResult importDatabase(BufferedReader reader) throws IOException {
         ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
         CffFormat citation = mapper.readValue(reader, CffFormat.class);
+        List<BibEntry> entriesList = new ArrayList<>();
+
+        // Retrieve mappings from CFF to JabRef
+        HashMap<String, Field> fieldMap = getFieldMappings();
+        HashMap<String, EntryType> typeMap = getTypeMappings();
+
+        // Parse main entry
         HashMap<Field, String> entryMap = new HashMap<>();
-        StandardEntryType entryType = StandardEntryType.Software;
+        EntryType entryType = typeMap.getOrDefault(citation.values.get("type"), StandardEntryType.Software);
 
         // Map CFF fields to JabRef Fields
-        HashMap<String, Field> fieldMap = getFieldMappings();
         for (Map.Entry<String, String> property : citation.values.entrySet()) {
             if (fieldMap.containsKey(property.getKey())) {
                 entryMap.put(fieldMap.get(property.getKey()), property.getValue());
-            } else if ("type".equals(property.getKey())) {
-                if ("dataset".equals(property.getValue())) {
-                    entryType = StandardEntryType.Dataset;
-                }
             } else if (getUnmappedFields().contains(property.getKey())) {
```

Comment:
Why not just collecting all non-handled fields here? Why to maintain a list of all unmapped fields? What if the specification updates?

Reply:
You are right, there is no sense to collect them here. This only makes sense in the exporter, to check if the `UnknownField` corresponds to one of CFF fields.